src/deploy.ts

```diff
@@ -77,31 +78,66 @@ type DeployTargetInfo =
   | {create: false; workspace: {id: string; login: string}; project: GetProjectResponse};
 
 /** Deploy a project to ObservableHQ */
-export async function deploy(
-  {config, message, force, deployPollInterval = DEPLOY_POLL_INTERVAL_MS, maxConcurrency}: DeployOptions,
-  effects = defaultEffects
-): Promise<void> {
-  const {clack} = effects;
-  Telemetry.record({event: "deploy", step: "start", force});
-  clack.intro(`${inverse(" observable deploy ")} ${faint(`v${process.env.npm_package_version}`)}`);
+export async function deploy(deployOptions: DeployOptions, effects = defaultEffects): Promise<void> {
+  Telemetry.record({event: "deploy", step: "start", force: deployOptions.force});
+  effects.clack.intro(`${inverse(" observable deploy ")} ${faint(`v${process.env.npm_package_version}`)}`);
 
-  let apiKey = await effects.getObservableApiKey(effects);
-  const apiClient = new ObservableApiClient(apiKey ? {apiKey, clack} : {clack});
-  const deployConfig = await effects.getDeployConfig(config.root);
-
-  if (deployConfig.workspaceLogin && !deployConfig.workspaceLogin.match(/^@?[a-z0-9-]+$/)) {
-    throw new CliError(
-      `Found invalid workspace login in ${join(config.root, ".observablehq", "deploy.json")}: ${
-        deployConfig.workspaceLogin
-      }.`
-    );
-  }
-  if (deployConfig.projectSlug && !deployConfig.projectSlug.match(/^[a-z0-9-]+$/)) {
-    throw new CliError(
-      `Found invalid project slug in ${join(config.root, ".observablehq", "deploy.json")}: ${deployConfig.projectSlug}.`
-    );
+  let deployInfo;
+  if (deployOptions.deployId) {
+    deployInfo = await continueExistingDeploy(deployOptions, effects, deployOptions.deployId);
+  } else {
+    deployInfo = await startNewDeploy(deployOptions, effects);
   }
```

Comment:
The usual style around here is to use a ternary to make it clear that both branches assign a value:

```suggestion
  const deployInfo = deployOptions.deployId 
    ? await continueExistingDeploy(deployOptions, effects, deployOptions.deployId)
    : await startNewDeploy(deployOptions, effects);
```